sdk/core/scheduler/main.cc

```diff
@@ -258,6 +258,15 @@ namespace sched
 	                                               size_t        mepc,
 	                                               size_t        mtval)
 	{
+#if !defined(NDEBUG)
+		/* Ensure that we got here from an IRQ-s deferred context */
+		void *returnAddress = __builtin_return_address(0);
+		Debug::Assert(
+		  __builtin_cheri_type_get(returnAddress) ==
```

Comment:
This can use Capability::type, which you can do without a cast if you make the 263 a Capability instead of a bare pointer.